spark-atlas-connector/src/main/scala/com/hortonworks/spark/atlas/SparkAtlasStreamingQueryEventTracker.scala

```diff
@@ -50,30 +48,25 @@ class SparkAtlasStreamingQueryEventTracker(
 
   override def onQueryProgress(event: QueryProgressEvent): Unit = {
     logInfo(s"Track running Spark Streaming query in the Spark Atlas: $event")
-    if (!streamQueryHashset.contains(event.progress.runId)) {
-      val query = SparkSession.active.streams.get(event.progress.id)
-      if (query != null) {
-        query match {
-          case query: StreamingQueryWrapper =>
-            val qd = QueryDetail.fromStreamingQueryListener(query.streamingQuery, event)
-            executionPlanTracker.pushEvent(qd)
-            streamQueryHashset.add(event.progress.runId)
+    val query = SparkSession.active.streams.get(event.progress.id)
```

Comment:
Can we relay StreamingQueryProgress.sources (i.e. the source progress) to decide whether to process the event ? i.e if the number of sources changed or the topic/partition data in the offset changed (for Kafka) ? Its tricky but thinking if it can avoid parsing the plan each time.

Reply:
I agree it would be efficient, but I don't prefer relying on string parse which we should do for extracting informations from `StreamingQueryProgress.sources` - for sink I had to do this because DSv1 sink doesn't have its logical plan.

I'm OK to apply this if we could just blindly compare the "sets" of source descriptions (this means we get rid of numbers which would be changed for each batch) between batches. I guess it would be incomplete, so put this in first level of check, and keep the newly added check as second level of check. What do you think?

Reply:
I just realized my idea cannot be applied, since descriptions in `StreamingQueryProcess.sources` will provide same even only part of sources are activated in this batch.

This is the set of descriptions in `StreamingQueryProcess.sources` when the micro-batch only activates one source, `sparkread0`:

```
sources

0 = "KafkaV2[SubscribePattern[sparkread0[1-3]]]"
1 = "KafkaV2[Subscribe[sparkread04, sparkread05]]"
2 = "KafkaV2[Assign[sparkread06-0, sparkread06-1, sparkread06-2, sparkread06-3, sparkread06-4, sparkread06-5, sparkread06-6, sparkread06-7, sparkread06-8, sparkread06-9, sparkread07-0, sparkread07-1, sparkread07-2, sparkread07-3, sparkread07-4, sparkread07-5, sparkread07-6, sparkread07-7, sparkread07-8, sparkread07-9]]"
```

So we should use same approach for checking whether we can skip processing or not, and populating sources.

One thing you may also notice from the above string representation - we can't get exact topic list from subscribe pattern if we deal with logical plan or `StreamingQueryProcess.sources`. (`[1-3]` is easy pattern but you may want to provide `*` or `?` or whatever which we can't populate all possible topics.) 

Hopefully, SAC deals with executed physical plan which contains actual Input(RDD)Partition to read for batch/streaming query - which is closer to track actual impacts which query is introducing. We may not want to lose this functionality.

Reply:
I think you have to look at the startOffset and endOffset (inside SourceProgress) which will have the expanded topic and list of partitions. Yes its kind of tricky and maybe it will work only for Kafka. (e.g. If there are some other sources that stores offsets differently we have to invent a new way). If its not straightforward, ok to parse the plan.

Reply:
@arunmahadevan 

> I think you have to look at the startOffset and endOffset (inside SourceProgress) which will have the expanded topic and list of partitions.

Yeah thanks for the info. I've missed this. I'd say that we rely on too many parts (logical plan, physical plan, and sink for streaming query status) and adding more sounds to have more chance to be broken, but the approach looks better to determine whether the plan should be re-examined or not.

Given you have idea here, would I ask you to find the time to see whether this approach could be applied to some other sources as well? `file` and `kafka` would be enough if we use that as earlier check and don't get rid of current check. Thanks in advance!